Ambiguous glob re-exports in `hal/src/serial/mod.rs`

Reported on CI and on my computer after `rustc` upgrade:
```
error: ambiguous glob re-exports
  --> hal/src/serial/mod.rs:23:9
   |
23 | pub use uart::*;
   |         ^^^^^^^ the name `Tx` in the type namespace is first re-exported here
...
26 | pub use usart::*;
   |         -------- but the name `Tx` in the type namespace is also re-exported here
   |
   = note: `-D ambiguous-glob-reexports` implied by `-D warnings`

error: ambiguous glob re-exports
  --> hal/src/serial/mod.rs:23:9
   |
23 | pub use uart::*;
   |         ^^^^^^^ the name `Rx` in the type namespace is first re-exported here
...
26 | pub use usart::*;
   |         -------- but the name `Rx` in the type namespace is also re-exported here

error: ambiguous glob re-exports
  --> hal/src/serial/mod.rs:23:9
   |
23 | pub use uart::*;
   |         ^^^^^^^ the name `Uart` in the type namespace is first re-exported here
...
26 | pub use usart::*;
   |         -------- but the name `Uart` in the type namespace is also re-exported here

error: ambiguous glob re-exports
  --> hal/src/serial/mod.rs:23:9
   |
23 | pub use uart::*;
   |         ^^^^^^^ the name `UartError` in the type namespace is first re-exported here
...
26 | pub use usart::*;
   |         -------- but the name `UartError` in the type namespace is also re-exported here

error: ambiguous glob re-exports
  --> hal/src/serial/mod.rs:23:9
   |
23 | pub use uart::*;
   |         ^^^^^^^ the name `Event` in the type namespace is first re-exported here
...
34 | pub use spi::*;
   |         ------ but the name `Event` in the type namespace is also re-exported here

error: ambiguous glob re-exports
  --> hal/src/serial/mod.rs:26:9
   |
26 | pub use usart::*;
   |         ^^^^^^^^ the name `Client` in the type namespace is first re-exported here
...
34 | pub use spi::*;
   |         ------ but the name `Client` in the type namespace is also re-exported here

error: ambiguous glob re-exports
  --> hal/src/serial/mod.rs:26:9
   |
26 | pub use usart::*;
   |         ^^^^^^^^ the name `Spi` in the type namespace is first re-exported here
...
34 | pub use spi::*;
   |         ------ but the name `Spi` in the type namespace is also re-exported here

error: ambiguous glob re-exports
  --> hal/src/serial/mod.rs:26:9
   |
26 | pub use usart::*;
   |         ^^^^^^^^ the name `SpiError` in the type namespace is first re-exported here
...
34 | pub use spi::*;
   |         ------ but the name `SpiError` in the type namespace is also re-exported here

error: could not compile `atsamx7x-hal` (lib) due to 8 previous errors
Error: Process completed with exit code 101.
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -45,6 +45,7 @@ jobs:
     strategy:
       matrix:
         pac: ${{ fromJson(needs.setup.outputs.pac_matrix) }}
+    continue-on-error: true # we let this fail and carry on, but GH UI is poor not showing amber like other CIs
     steps:
       - name: Checkout sources
         uses: actions/checkout@v2

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -1,5 +1,5 @@
 name: HAL
-on: [push]
+on: [push, pull_request]
 
 env:
   RUSTDOCFLAGS: -D warnings
@@ -8,6 +8,7 @@ env:
 
 jobs:
   setup:
+    if: github.event.pull_request.head.repo.full_name != github.repository
     runs-on: ubuntu-latest
     steps:
       - name: Checkout sources
@@ -24,6 +25,7 @@ jobs:
       feature_matrix: ${{ steps.features.outputs.feature_matrix }}
 
   build:
+    if: github.event.pull_request.head.repo.full_name != github.repository
     needs: setup
     runs-on: ubuntu-latest
     strategy:
@@ -40,6 +42,7 @@ jobs:
         run: cargo check --package atsamx7x-hal --features ${{ matrix.pac }},unproven
 
   build-latest:
+    if: github.event.pull_request.head.repo.full_name != github.repository
     needs: setup
     runs-on: ubuntu-latest
     strategy:
@@ -59,6 +62,7 @@ jobs:
         run: cargo check --package atsamx7x-hal --features ${{ matrix.pac }},unproven
 
   build-features:
+    if: github.event.pull_request.head.repo.full_name != github.repository
     needs: setup
     runs-on: ubuntu-latest
     strategy:
@@ -75,6 +79,7 @@ jobs:
         run: cargo check --package atsamx7x-hal --features ${{ matrix.features }},samv71q21b,unproven
 
   board-examples:
+    if: github.event.pull_request.head.repo.full_name != github.repository
     needs: setup
     runs-on: ubuntu-latest
     strategy:
@@ -93,6 +98,7 @@ jobs:
           cargo check --examples
 
   clippy-hal:
+    if: github.event.pull_request.head.repo.full_name != github.repository
     needs: build
     runs-on: ubuntu-latest
     steps:
@@ -108,6 +114,7 @@ jobs:
         run: cargo clippy --package atsamx7x-hal --no-deps --features samv71q21b,unproven,reconfigurable-system-pins -- --deny warnings
 
   clippy-examples:
+    if: github.event.pull_request.head.repo.full_name != github.repository
     needs: [setup, build, board-examples]
     runs-on: ubuntu-latest
     strategy:
@@ -128,6 +135,7 @@ jobs:
           cargo clippy --examples --no-deps -- --deny warnings
 
   docs:
+    if: github.event.pull_request.head.repo.full_name != github.repository
     needs: build
     runs-on: ubuntu-latest
     steps:

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -39,6 +39,24 @@ jobs:
       - name: Build HAL for ${{ matrix.pac }}
         run: cargo check --package atsamx7x-hal --features ${{ matrix.pac }},unproven
 
+  build-latest:
+    needs: setup
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        pac: ${{ fromJson(needs.setup.outputs.pac_matrix) }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - name: Update toolchain
+        run: rustup update
+      - name: Specify toolchain
+        run: rustup override set stable
+      - name: Install Rust (thumbv7em)
+        run: rustup target add thumbv7em-none-eabihf
+      - name: Build HAL for ${{ matrix.pac }}
+        run: cargo check --package atsamx7x-hal --features ${{ matrix.pac }},unproven
+
   build-features:
     needs: setup
     runs-on: ubuntu-latest

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -1,5 +1,5 @@
 name: HAL
-on: [push, pull_request]
+on: [push]
 
 env:
   RUSTDOCFLAGS: -D warnings

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,8 +4,7 @@
 
 ### Added
 
-- [Integration](https://crates.io/crates/mcan-core) with the
-[`mcan`](https://crates.io/crates/mcan) crate.
+- [Integration](https://crates.io/crates/mcan-core) with the [`mcan`](https://crates.io/crates/mcan) crate.
 - Implementation of blocking::i2c::Transactional trait from [embedded-hal](https://crates.io/crates/embedded-hal) for TWI device.
 
 ### Changed
@@ -14,6 +13,7 @@
 ### Removed
 
 ### Fixed
+- [#62] Remove ambiguous reexports from `src/serial/mod.rs`.
 
 ## [v0.4.2] 2022-11-06
 

--- a/hal/src/serial/mod.rs
+++ b/hal/src/serial/mod.rs
@@ -20,15 +20,11 @@ impl ExtBpsU32 for u32 {
 }
 
 pub mod uart;
-pub use uart::*;
 
 pub mod usart;
-pub use usart::*;
 
 pub mod twi;
 pub use twi::*;
 
 #[cfg(not(feature = "pins-64"))]
 pub mod spi;
-#[cfg(not(feature = "pins-64"))]
-pub use spi::*;

--- a/hal/src/serial/usart/uart.rs
+++ b/hal/src/serial/usart/uart.rs
@@ -14,7 +14,7 @@ use crate::ehal::{self, blocking};
 pub enum UartError {
     /// A frame's stop bit was read as zero.
     Framing,
-    /// Parity check of the recived frame failed.
+    /// Parity check of the received frame failed.
     Parity,
     /// A word was received before the previous word was read.
     Overrun,

